src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Serialization/ObjectWriter.cs

```diff
@@ -278,25 +311,34 @@ public void WriteValue(ReadOnlySpan<byte> span)
                     break;
             }
 
-            var elementType = typeof(byte);
-            Debug.Assert(s_typeMap[elementType] == TypeCode.UInt8);
+            var elementType = typeof(T);
+            Contract.ThrowIfFalse(s_typeMap.ContainsKey(elementType));
```

Comment:
>s_typeMap [](http://example.com/codeflow?start=34&length=9)

TryGetValue to avoid duplicated lookup?

Reply:
i don't mind this too much.  so we get a nice contract failure if we defy expectations here.

Reply:
Not a big deal, but not sure why this is less clear:

Contract.ThrowIfFalse(s_typeMap.TryGetValue(elementType, out var actualType));
Contract.ThrowIfTrue(expectedType != actualType);